Remove incorrect 'when dark' condition example

janick opened the pull request:
## Proposed change
<!-- 
    'when dark' example was explicitly described earlier as incorrect, then shown correctly a little lower.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

janick committed:
Remove incorrect 'when dark' condition example

homeassistant commented:
Hi @janick, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#23591).

Once you do that we will be able to review and accept this pull request.

Thanks!

frenck commented on source/_docs/scripts/conditions.markdown:
````diff
@@ -383,28 +383,18 @@ Note that if only `before` key is used, the condition will be `true` _from midni
 
-This is 'when dark' - equivalent to a state condition on `sun.sun` of `below_horizon`:
-
-```yaml
-condition:
-  - condition: sun
-    after: sunset
-    before: sunrise
-```
````
This not functioning is a bug, not a documentation error.

janick replied:
I did not check if this was working as it does not follow the documented principle that all conditions must be true at the same time. There is no way you can be before sunrise AND after sunset on the same day. I'll give it a try tonite.

I agree it would be nicer, but it could complicate the interpretation of multiple conditions. But if it works as a special case, then the documentation still needs to be updated (differently) to eliminate the contradiction.

janick replied:
Well, it may be a bug, but it does not mean we should document what does not work.... Updated PR to indicate that particular condition does not work as 2022.07.7

https://github.com/home-assistant/core/issues/70781

frenck replied:
We don't document references to bugs in our documentation. Please revert these changes

janick replied:
Updated to reflect intent. Suggested fix here: https://github.com/home-assistant/core/issues/70781#issuecomment-1201481041

janick replied:
Also, I find "not documenting references to bugs" rather silly.

Documentation is most useful when it reflects what is, not what should be.

frenck replied:
> Documentation is most useful when it reflects what is, not what should be.

That is true, the reality is though, that workarounds and temporary things to work with issue tent to stay in the documentation as well. Which makes it all really hard to maintain as well. Hence, we don't document bugs.

janick replied:
Why are you closing this? The contradiction still remains.

You can't have both "the condition will be `true` _from midnight_ until sunrise/sunset. If only `after` key is used, the condition will be `true` from sunset/sunrise _until midnight_ (...) When both keys are used, the result is a logical `and` of separate conditions" and the "- condition: sun
    after: sunset
    before: sunrise" example!

You need to either accept the PR as-is and remove the contradiction (and work-around to use the 'or' operator), or accept the original PR that simply removed the example that does not work.

janick committed:
Update conditions.markdown

janick committed:
Removed reference to bug

Documenting only what actually works.

janick committed:
Remove contradictions

This is the apparent intent and should work if the patch suggested here is added

janick committed:
Fixed missing words

frenck commented:
Closing this PR as it document an issue.

janick reviewed (commented):
Contradiction remains.
